### PR TITLE
feat(an): Illustrated radio button

### PR DESCRIPTION
**Pull request by Thorrik58**
# Adding prop and styling to radio button to add an illustration according to design.

Attach a link to issue if relevant

## What

Adding prop and styling to radio button to add an illustration according to design.
Also fix to error message for cases when tooltip and/or illustration is present on radio button.

## Why

Needed for the accident notification application.

## Screenshots / Gifs

**Default state** 

### Desktop:
image

### Mobile:
image

### Half width Desktop:
image

### Half width Mobile:
image

---

**Error Message state**

**Before** changes to error message, messages will appear like so:
### Large:
image

### Small:
image

---

**After** changes to error message and illustration added will appear so:
### Large Desktop:
image

### Large Mobile:
image

### Small Desktop:
image

### Small Mobile:
image

### Half width Large Desktop:
image

### Half width Large Mobile:
image

### Half width Small Desktop:
image

### Half width Small Mobile:
image

---

This is another approach which I liked a little bit more but didnt not implement as its further away from the original, would love some input.
### Large:
image

### Small:
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by andriandresson — COMMENTED** [review]
The illustration addition to the radioButton looks good. 
I have few  thoughts about the error state, maybe it would be enough to have a single error state for the whole radioButton instead of error state on every single option in the radioButton? 
**For example:**
image

**Review by eirikurn — APPROVED** [review]
Approving Aranja files.

Would like to get Island UI input on this though. At least tweak the spacing in the small variant so the error message aligns with ... something :D

**Comment by Thorrik58** [comment]
Good point, I fixed the spacing so it aligns on the small version.

File: libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx [hunk 1]
```diff
@@ -109,25 +114,28 @@ export const RadioButton = ({
...
+            <Illustration />
+          </Box>
+        )}
         {tooltip && (
           <div
             className={cn(styles.tooltipContainer, {
               [styles.tooltipLargeContainer]: large,
+              [styles.toolTipLargeContainerWithIllustration]:
+                large && Illustration,
             })}
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx:128]: this should probably be separated so no styles are overwritten, if large is true and/or Illustration is true marginLeft will conflict between all classnames, this makes the styles more fragile

File: libs/application/templates/accident-notification/src/assets/WorkTypeIllustration.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
+import React, { FC } from 'react'
+
+export const WorkTypeIllustration: FC = () => (
```
Comment (barabrian) [inline libs/application/templates/accident-notification/src/assets/WorkTypeIllustration.tsx:3]: We should probably avoid unnecessary use of RC https://github.com/island-is/island.is/pull/3621#discussion_r620690786

**Review by barabrian — COMMENTED** [review]
LGTM, just some minor comments

**Review by barabrian — APPROVED** [review]
LGTM

**Review by Toti91 — APPROVED** [review]
LGTM 👍

**Pull request merged**